app/services/arcgis_api/geocoder.rb

```diff
@@ -96,7 +96,23 @@ def find_address_candidates(**options)
     # it expires after 1 hour
     # @return [String] Auth token
     def retrieve_token!
-      token, expires = request_token.fetch_values('token', 'expires')
+      begin
+        token, expires = request_token.fetch_values('token', 'expires')
+      rescue => err
+        token = 'invalidtoken'
+        expires = 1000
```

Comment:
will this write a bad token to the cache? what if we deleted whatever is there, instead of saving a bad value that another process could pick up and try to use?

Reply:
Great suggestion. I'll add that to some other fixes.